app/controllers/channel_transfer_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ChannelTransferController < ApplicationController
+  include ChannelsHelper
+
+  before_action :verify_token
+
+  attr_reader :current_channel
+
+  def reject_transfer
+    Channels::RejectChannelTransfer.new(channel: current_channel).perform
+    redirect_to home_publishers_path, notice: t("shared.channel_transfer_rejected")
+  end
+
+  private
+
+  def verify_token
+    @current_channel = Channel.find(params[:id])
+    if @current_channel.nil? || @current_channel.contest_token.empty? || @current_channel.contest_token != params[:token_id]
```

Comment:
I believe we should delete the token as a way of invalidating to prevent someone from trying to steal with multiple tokens.

Reply:
@yachtcaptain23 the token is set to `nil` in `Channels::RejectChannelTransfer`